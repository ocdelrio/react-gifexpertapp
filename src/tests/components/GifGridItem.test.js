import React from 'react';
import {shallow} from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem'
import '@testing-library/jest-dom';


describe('Test GifGridItem', () => {

  const urlTest = "https://www.test.test";
  const titleTest = "titulo";
  const wrapper = shallow(<GifGridItem url={urlTest} title={titleTest}/>);

  test('Test', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de tener un pÃ¡rrafo con el tittle', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(titleTest);
  });

  test('Debe tener una imagen con alt y source', () => {
    const img = wrapper.find('img');
    expect(img.props().src).toBe(urlTest);
    expect(img.prop('alt')).toBe(titleTest);
  });


  test('Debe tener las classes asignadas de animacion', () => {
    const div = wrapper.find('div');
    expect(div.hasClass("card animate__animated animate__fadeIn")).toBe(true);
  });
  
  
  
  
});

