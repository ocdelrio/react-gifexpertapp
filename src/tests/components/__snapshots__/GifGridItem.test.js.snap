// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifGridItem Test 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="titulo"
    src="https://www.test.test"
  />
  <p>
    titulo
  </p>
</div>
`;
